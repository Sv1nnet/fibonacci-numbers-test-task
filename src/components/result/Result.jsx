import React from 'react';
import PropTypes from 'prop-types';

import './result.scss';

const Result = ({ result, error }) => (
  <div className="results-container">
    <h3 className="center-align">Result:</h3>
    <h4 className={`center-align result-text ${error ? 'red-text accent-4' : ''}`}>{result}</h4>
  </div>
);

Result.propTypes = {
  result: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
  error: PropTypes.bool,
};

Result.defaultProps = {
  result: null,
  error: false,
};

export default Result;
