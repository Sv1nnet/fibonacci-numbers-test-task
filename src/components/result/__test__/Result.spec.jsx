import { it, describe, expect } from '@jest/globals';
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Result from '../Result';

configure({ adapter: new Adapter() });

describe('Test Result component', () => {
  it('Should match snapshot with empty props', () => {
    const component = shallow(<Result />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Should render result inside <h4> in Result', () => {
    const result = 154;
    const component = shallow(<Result result={result} />);

    expect(component.find('h4').text()).toBe(result.toString());
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Should render as an error message inside <h4> and have error class in Result', () => {
    const result = 'Error occured';
    const errorClassesRegexp = /\s(red-text accent-4)(\s|$)/;
    const component = shallow(<Result error result={result} />);

    expect(component.find('h4').hasClass(errorClassesRegexp)).toBe(true);
    expect(component.find('h4').text()).toBe(result);
    expect(toJson(component)).toMatchSnapshot();
  });
});
