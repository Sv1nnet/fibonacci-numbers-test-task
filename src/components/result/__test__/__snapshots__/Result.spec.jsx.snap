// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Result component Should match snapshot with empty props 1`] = `
<div
  className="results-container"
>
  <h3
    className="center-align"
  >
    Result:
  </h3>
  <h4
    className="center-align result-text "
  />
</div>
`;

exports[`Test Result component Should render as an error message inside <h4> and have error class in Result 1`] = `
<div
  className="results-container"
>
  <h3
    className="center-align"
  >
    Result:
  </h3>
  <h4
    className="center-align result-text red-text accent-4"
  >
    Error occured
  </h4>
</div>
`;

exports[`Test Result component Should render result inside <h4> in Result 1`] = `
<div
  className="results-container"
>
  <h3
    className="center-align"
  >
    Result:
  </h3>
  <h4
    className="center-align result-text "
  >
    154
  </h4>
</div>
`;
